src/gluonts/model/seasonal_naive/_predictor.py

```diff
@@ -70,16 +70,11 @@ def __init__(
             else get_seasonality(freq)
         )
 
-    def predict(self, dataset: Dataset, **kwargs) -> Iterator[SampleForecast]:
-        for data in dataset:
-            start = pd.Timestamp(data["start"], freq=self.freq)
-            target = np.asarray(data["target"], np.float32)
-            yield self._predict_time_series(start_time=start, target=target)
-
-    def _predict_time_series(
-        self, start_time: pd.Timestamp, target: np.ndarray
-    ) -> SampleForecast:
+    def predict_item(self, item: DataEntry) -> Forecast:
+        target = np.asarray(item["target"], np.float32)
```

Comment:
Target is already np.array

Reply:
Yes, but in this case the explicit casting to `np.float32` is necessary, so that when we calculate the OWA we get exactly `1.0` when testing the `Naive2Predictor`.